internal/wasm/store_module_list.go

```diff
@@ -29,6 +29,22 @@ func (s *Store) deleteModule(m *ModuleInstance) error {
 
 	if m.ModuleName != "" {
 		delete(s.nameToModule, m.ModuleName)
+
+		// shrink the map if it's allocated more than twice the size of the list
+		nameToModuleLen := len(s.nameToModule)
+		if s.nameToModuleCap > nameToModuleShrinkThreshold && s.nameToModuleCap >= nameToModuleLen*2 {
```

Comment:
Say `s.nameToModuleCap == 101` and `nameToModuleLen == 50`. The test passes, and we enter the if to resize the map from `101`, to `100` (`nameToModuleShrinkThreshold`). We'll have 50 slots of spare capacity, but it hardly seems worth it to resize.

One fix is to do:
```go
if s.nameToModuleCap > nameToModuleShrinkThreshold*2 && s.nameToModuleCap >= nameToModuleLen*2 {
```

Another, which I personally find clearer is:
```go
newCap := len(s.nameToModule)
if newCap < nameToModuleShrinkThreshold {
	newCap = nameToModuleShrinkThreshold
}
if newCap*2 <= s.nameToModuleCap {
	// do the resize
}
```